### PR TITLE
Additional network related metric logs

### PR DESCRIPTION
**Pull request by greymistcube**
Adds logs for counting sockets in use and request process duration.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ Version 0.26.4
 
 To be released.
 
+ -  Additional logs tagged `Metric` to monitor `ITransport` behavior.  [#1802]
```
Comment (dahlia) [inline CHANGES.md:9]:
```suggestion
 -  Additional logs tagged `Metric` to monitor `ITransport` behavior.  [[#1802]]
```

File: Libplanet/Net/Transports/NetMQTransport.cs [hunk 2]
```diff
@@ -813,96 +815,123 @@ private async Task ProcessRequest(MessageRequest req, CancellationToken cancella
...
-                );
+                long incrementedSocketCount = Interlocked.Increment(ref _socketCount);
+                _logger
+                    .ForContext("Tag", "Metric")
+                    .Debug(
+                    "{SocketCount} sockets open for processing request {Message} {RequestId}.",
+                    incrementedSocketCount,
+                    req.Message,
+                    req.Id);
+                using var dealer = new DealerSocket(req.Peer.ToNetMQAddress());
```
Comment (riemannulus) [inline Libplanet/Net/Transports/NetMQTransport.cs:828]: Is it intended to `DealerSocket` move into `try` block?

File: Libplanet/Net/Transports/NetMQTransport.cs [hunk 3]
```diff
@@ -44,6 +44,7 @@ public class NetMQTransport : ITransport
 
         private Channel<MessageRequest> _requests;
         private long _requestCount;
+        private long _socketCount;
```
Comment (dahlia) [inline Libplanet/Net/Transports/NetMQTransport.cs:47]: How about grouping fields used only for logging, and leaving a comment for these fields?

```suggestion
        // Used only for logging:
        private long _requestCount;
        private long _socketCount;

        private Channel<MessageRequest> _requests;
```

File: Libplanet/Net/Transports/NetMQTransport.cs (hunk 2 above)
Comment (greymistcube) [reply Libplanet/Net/Transports/NetMQTransport.cs]: This is the **dumbest** way to ensure `_socketCount` stays valid. 🙄
Had problems `_socketCount` becoming negative for some reason.
I agree, structurally it's a mess. I've made another attempt to fix this.

Comment (greymistcube) [reply Libplanet/Net/Transports/NetMQTransport.cs]: See the updated changes.

File: CHANGES.md (hunk 1 above)
Comment (greymistcube) [reply CHANGES.md]: Fixed. 😥

File: Libplanet/Net/Transports/NetMQTransport.cs (hunk 3 above)
Comment (greymistcube) [reply Libplanet/Net/Transports/NetMQTransport.cs]: Fixed. See the updated file changes.

**Review by dahlia — DISMISSED** [review]

**Review by dahlia — APPROVED** [review]

**Pull request merged**